tests/unit/lms/services/file_test.py

```diff
@@ -24,6 +26,58 @@ def test_it_doesnt_return_matching_files_from_other_application_instances(
 
         assert not svc.get(file_.lms_id, file_.type)
 
+    def test_upsert(self, db_session, svc, application_instance):
+        existing_files_count = db_session.query(File).count()
+
+        # This files will be created on the DB
+        update_files = factories.File.create_batch(
+            5, application_instance=application_instance
+        )
+        db_session.flush()
+        # These won't exist on the DB
+        insert_files = factory_boy.build_batch(dict, 5, FACTORY_CLASS=factories.File)
```

Comment:
`File.build_batch` objects ended up in the DB.

Reply:
Were they really ending up in the DB? `factories.File.build_batch()` definitely isn't supposed to add things to the DB, and when I tested it didn't seem to, this seems to work:

```diff
diff --git a/tests/unit/lms/services/file_test.py b/tests/unit/lms/services/file_test.py
index f020defe..bf73b2e5 100644
--- a/tests/unit/lms/services/file_test.py
+++ b/tests/unit/lms/services/file_test.py
@@ -35,7 +35,7 @@ class TestGet:
         )
         db_session.flush()
         # These won't exist on the DB
-        insert_files = factory_boy.build_batch(dict, 5, FACTORY_CLASS=factories.File)
+        insert_files = factories.File.build_batch(5)
         assert db_session.query(File).count() == existing_files_count + len(
             update_files
         )
@@ -53,9 +53,9 @@ class TestGet:
             ]
             + [
                 {
-                    "type": file["type"],
-                    "course_id": file["course_id"],
-                    "lms_id": file["lms_id"],
+                    "type": file.type,
+                    "course_id": file.course_id,
+                    "lms_id": file.lms_id,
                     "name": f"insert_file_{i}",
                     "size": i * 100,
                 }
@@ -73,8 +73,8 @@ class TestGet:
             assert file.size == i * 10
             assert file.name == f"update_file_{i}"
 
-        for i, file_dict in enumerate(insert_files):
-            file = db_session.query(File).filter_by(lms_id=file_dict["lms_id"]).one()
+        for i, file_ in enumerate(insert_files):
+            file = db_session.query(File).filter_by(lms_id=file_.lms_id).one()
             assert file.size == i * 100
             assert file.name == f"insert_file_{i}"
  
```

Reply:
> Were they really ending up in the DB? 

They were at some point, it took me a while to find the `factory_boy.build_batch(dict, 5, FACTORY_CLASS=factories.File)` workaround. But seems to work now? :shrug: